fastjet/internal/delaunay/triangulation.go

```diff
@@ -0,0 +1,1066 @@
+// Copyright 2017 The go-hep Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package delaunay
+
+import (
+	"fmt"
+	"math/rand"
+	"time"
+)
+
+// Delaunay holds necessary information for the
+// delaunay triangulation
+type Delaunay struct {
+	// triangles is a slice of all triangles that have been created. It is used to get the final
+	// list of triangles in the delaunay triangulation
+	triangles triangles
+	// root is a triangle that contains all points. It is used as the starting point in the hierarchy
+	// to find the triangle that contains a point
+	root                   *Triangle
+	maxX, minX, maxY, minY float64
+	// hierarchy indicates which method to use to find the triangle that contains the point
+	hierarchy bool
+}
+
+// HierarchicalDelaunay creates a delaunay triangulation with the given points
+// all points have to be inside the user defined bounds
+// uses a hierarchy to find the triangle which contains a point
+// has a worst time complexity of O(nln(n))
+func HierarchicalDelaunay(points []*Point, maxX float64, minX float64, maxY float64, minY float64) *Delaunay {
```

Comment:
Why was this renamed? `NewDelaunay` works fine for me.

Also note that the params can be `(pts []*Point, maxX, minX, maxY, minY float64) *Delaunay`.